lib.ml

```diff
@@ -680,35 +703,67 @@ let env : (string * (located_primitive * Types.datatype * pure)) list = [
                                            | (`XML x) -> Value.value_of_xmlitem x
                                            | _ -> failwith "non-XML passed to xmlToVariant") xs)
                     | _ -> failwith "non-XML passed to xmlToVariant")),
-   datatype "(Xml) ~> mu n.[ [|Text:String | Attr:(String, String) | Node:(String, n) |] ]",
+  datatype "(Xml) ~> mu n.[ [|Text:String | Attr:(String, String) | Node:(String, n) | NsAttr: (String, String, String) | NsNode: (String, String, n) |] ]",
+  IMPURE);
+
+  "xmlItemToVariant",
+  (`Server (p1 (fun v ->
```

Comment:
Is there any reason that this only makes sense on the server?

I could imagine wanting to create XML on the client, without having to do a client call. If this is outside of the scope of the current PR then fair enough, but it would be good to have an issue requesting a client implementation as well.